### PR TITLE
Adds support for combined facets, refines styling

### PR DESCRIPTION
**Pull request by devonbush**
facets are wider, and more closely match the original design 
image

Support for combined facets in a single modal has been added
image

Also refactors the closeable modal code into a custom hook

### CODE REVIEW

**Comment by devonbush** [comment]
temporarily closing, found a bug that's going to require a bit of refactoring

File: app/javascript/components/search/controls/FiltersBoxSearchable.js [hunk 1]
```diff
@@ -25,8 +25,16 @@ export default function FiltersBoxSearchable({ facet, selection, setSelection, s
    * Form of IDs: <general name> <specific name(s)>
    * General: All lowercase, specified in app code (e.g. 'apply-facet')
    * Specific: Cased as specified in API (e.g. 'species', 'NCBItaxon9606')
-   *
+   * /single_cell/studies/5e9e07bd771a5b2caa140971/upload
```
Comment (eweitz) [inline app/javascript/components/search/controls/FiltersBoxSearchable.js:28]: I like having specific examples in code.  Having a bit more context in this example would help clarify its purpose.

File: app/javascript/components/search/controls/FiltersBoxSearchable.js [hunk 2]
```diff
@@ -25,8 +25,16 @@ export default function FiltersBoxSearchable({ facet, selection, setSelection, s
...
    * UI code concatenates names in the ID.  Names in ID are hyphen-delimited.
+
```
Comment (eweitz) [inline app/javascript/components/search/controls/FiltersBoxSearchable.js:30]: Similarly, this specific example looks helpful, but more context in the comment would clarify.

File: app/javascript/hooks/closeableModal.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+  Example usage might be:
+
+  const [show, setShow] = useState(false)
+  const { node, clearNode, handleButtonClick } = useCloseableModal(show, setShow)
+  <span ref={node}>
+    <a onClick={handleButtonClick}>Toggle thingy</a>
+    <a ref={clearNode}>Close thingy</a>
+    { show && <thingy/> }
+  </span>
```
Comment (eweitz) [inline app/javascript/hooks/closeableModal.js:21]: Very helpful!

File: app/javascript/hooks/closeableModal.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+ */
+export default function useCloseableModal(show, setShow) {
+   const clearNode = useRef()
+  /**
+    * Click on the facet control itself
+    */
```
Comment (eweitz) [inline app/javascript/hooks/closeableModal.js:27]: A few months ago you suggested single-line IDE-hookable docstrings (e.g. `/** Click on the facet control itself */`) for such brief comments.  I've found that useful and would do so here and in `clearFacet`.

**Pull request closed without merging**